qiskit_research/utils/gate_decompositions.py

```diff
@@ -28,13 +28,42 @@
     XXPlusYYGate,
 )
 from qiskit.dagcircuit import DAGCircuit
-from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+from qiskit.providers.backend import Backend
+from qiskit.pulse import ControlChannel, Play
 from qiskit.qasm import pi
 from qiskit.transpiler.basepasses import TransformationPass
 
 from .gates import SECRGate
 
 
+def cr_forward_direction(control, target, inst_sched_map, ctrl_chans) -> bool:
```

Comment:
The arguments still need type annotations. Also, it just occurred to me that the qubit arguments don't really have anything to do with controls or targets, they should just be called `qubit_1`, `qubit_2` or something.